File: scripts/merge_tasks.js
const fs = require('fs');

// Read the existing tasks
const existingTasks = JSON.parse(fs.readFileSync('tasks/tasks.json', 'utf8'));
// Read the new tasks
const newTasks = JSON.parse(fs.readFileSync('tasks/tasks.json.new', 'utf8'));

// Create a map of existing tasks by ID for quick lookup
const existingTasksMap = {};
existingTasks.tasks.forEach(task => {
  existingTasksMap[task.id] = task;
});

// Merge the tasks, preserving status and subtasks of existing tasks
const mergedTasks = {
  tasks: newTasks.tasks.map(newTask => {
    const existingTask = existingTasksMap[newTask.id];
    if (existingTask) {
      // Preserve status, subtasks, and other fields from existing task
      return {
        ...newTask,
        status: existingTask.status,
        subtasks: existingTask.subtasks || newTask.subtasks
      };
    }
    return newTask;
  })
};

// Write the merged tasks to a new file
fs.writeFileSync('tasks/tasks.json.merged', JSON.stringify(mergedTasks, null, 2));
console.log('Tasks merged successfully!');

File: scripts/merge_tasks.py
import json
import os

# Read the existing tasks
with open('tasks/tasks.json', 'r') as f:
    existing_tasks = json.load(f)

# Read the new tasks
with open('tasks/tasks.json.new', 'r') as f:
    new_tasks = json.load(f)

# Create a map of existing tasks by ID for quick lookup
existing_tasks_map = {task['id']: task for task in existing_tasks['tasks']}

# Merge the tasks, preserving status and subtasks of existing tasks
merged_tasks = {
    'tasks': [
        {
            **new_task,
            'status': existing_tasks_map.get(new_task['id'], {}).get('status', new_task['status']),
            'subtasks': existing_tasks_map.get(new_task['id'], {}).get('subtasks', new_task.get('subtasks', []))
        }
        for new_task in new_tasks['tasks']
    ]
}

# Write the merged tasks to a new file
with open('tasks/tasks.json.merged', 'w') as f:
    json.dump(merged_tasks, f, indent=2)

print('Tasks merged successfully!')
